### PR TITLE
Makefile: Properly rebuild po-docgen on src changes

### PR DESCRIPTION
**Pull request by mxinden**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -78,7 +78,7 @@ hack/prometheus-config-reloader-image: cmd/prometheus-config-reloader/Dockerfile
 ##############
 
 .PHONY: generate
-generate: pkg/client/monitoring/v1/zz_generated.deepcopy.go pkg/client/monitoring/v1/openapi_generated.go jsonnet/prometheus-operator/**-crd.libsonnet bundle.yaml kube-prometheus Documentation/*
+generate: pkg/client/monitoring/v1/zz_generated.deepcopy.go pkg/client/monitoring/v1/openapi_generated.go jsonnet/prometheus-operator/**-crd.libsonnet bundle.yaml kube-prometheus $(shell find Documentation -type f)
```
Comment (squat) [inline Makefile]: Now that you are replacing the splats with shell invocations, consider replacing `jsonnet/prometheus-operator/**-crd.libsonnet` as well?

Comment (mxinden) [reply Makefile]: @squat Good point, will follow up.

File: Makefile [hunk 2]
```diff
@@ -215,7 +215,7 @@ $(JB_BINARY):
 $(PO_CRDGEN_BINARY): cmd/po-crdgen/main.go pkg/client/monitoring/v1/openapi_generated.go
 	go install github.com/coreos/prometheus-operator/cmd/po-crdgen
 
-$(PO_DOCGEN_BINARY): cmd/po-docgen/**.go
+$(PO_DOCGEN_BINARY): $(shell find cmd/po-docgen -type f)
```
Comment (brancz) [inline Makefile]: I think this also has a dependency on `pkg/client/monitoring/v1/types.go`.

**Comment by mxinden** [comment]
@brancz @squat Thanks for the feedback. Could you take another look?

**Comment by brancz** [comment]
lgtm on green

**Comment by brancz** [comment]
There are some jsonnet changes that were just merged, which is why the first job fails, just re-generate locally and add an additional commit.

**Pull request merged**